src/frontend/components/DashboardDocument/index.spec.tsx

```diff
@@ -61,9 +61,10 @@ describe('<DashboardDocument />', () => {
 
     // First backend call: the document is still processing
     jest.advanceTimersByTime(1000 * 10 + 200);
-    await wait();
+    await waitFor(() =>
+      expect(fetchMock.lastCall()![0]).toEqual('/api/documents/44/'),
```

Comment:
For promises, I have found that controlling the exact flow of their resolution using a `Deferred` [(here's one if you're not familiar with the idea)](https://github.com/MTES-MCT/partaj/blob/master/src/frontend/js/utils/test/Deferred.ts) avoids the need to wait for anything.

You can just `await act(async () => deferred.resolve("whatever"))` when you want to trigger the promise being resolved. It's also pretty handy to test loading states, etc